File: RDBMS/Engine/Repl.cs
using System;
using System.Threading.Tasks;

namespace RDBMS.Engine
{
    public class Repl
    {
        private readonly Database _db;

        public Repl(Database db)
        {
            _db = db;
        }

        public void Start()
        {
            Task.Run(() => 
            {
                // Give the web server a second to start logs
                System.Threading.Thread.Sleep(1000); 
                Console.WriteLine("\n=================================");
                Console.WriteLine("   SQL-ly REPL READY");
                Console.WriteLine("   Try: INSERT INTO users VALUES (1, \"Admin\", 99)");
                Console.WriteLine("=================================\n");

                while (true)
                {
                    Console.Write("SQL> ");
                    var input = Console.ReadLine();
                    if (string.IsNullOrWhiteSpace(input)) continue;
                    if (input == "exit") break;

                    var result = _db.ExecuteSql(input);
                    Console.WriteLine(result);
                }
            });
        }
    }
}

File: sqlly-go/internal/engine/repl.go
package engine

import (
	"bufio"
	"fmt"
	"os"
	"strings"
	"time"
)

type Repl struct {
	Db *Database
}

func (r *Repl) Start() {
	go func() {
		time.Sleep(1 * time.Second)
		fmt.Println("\n=================================")
		fmt.Println("   SQLly REPL READY (Go Version)")
		fmt.Println(`   Try: INSERT INTO users VALUES (1, "Admin", 99)`)
		fmt.Println("=================================\n")

		scanner := bufio.NewScanner(os.Stdin)
		fmt.Print("SQLly> ")

		for scanner.Scan() {
			input := scanner.Text()
			if strings.TrimSpace(input) == "" {
				fmt.Print("SQLly> ")
				continue
			}
			if input == "exit" {
				break
			}

			result := r.Db.ExecuteSql(input)
			fmt.Println(result)
			fmt.Print("SQLly> ")
		}
	}()
}